src/posix/platform/radio_url.cpp

```diff
@@ -0,0 +1,209 @@
+
+#include <stdio.h>
+#include <string.h>
+
+#include <openthread/openthread-system.h>
+
+#include "core/common/code_utils.hpp"
+#include "posix/platform/platform-posix.h"
+
+const char *otSysGetRadioUrlHelpString(void)
+{
+#if OPENTHREAD_POSIX_CONFIG_RCP_BUS == OT_POSIX_RCP_BUS_SPI
+    return "Radio URL:\n"
+           "spinel+spi://${PATH_TO_SPI_DEVICE}?${arguments}\n"
+           "arguments:\n"
+           " gpio-int-device[=gpio-device-path]\n"
+           "                           Specify a path to the Linux sysfs-exported GPIO device for the\n"
+           "                           `I̅N̅T̅` pin. If not specified, `SPI` interface will fall back to\n"
+           "                           polling, which is inefficient.\n"
+           " gpio-int-line[=line-offset]\n"
+           "                           The offset index of `I̅N̅T̅` pin for the associated GPIO device.\n"
+           "                           If not specified, `SPI` interface will fall back to polling,\n"
+           "                           which is inefficient.\n"
+           " gpio-reset-dev[=gpio-device-path]\n"
+           "                           Specify a path to the Linux sysfs-exported GPIO device for the\n"
+           "                           `R̅E̅S̅` pin.\n"
+           " gpio-reset-line[=line-offset]"
+           "                           The offset index of `R̅E̅S̅` pin for the associated GPIO device.\n"
+           " spi-mode[=mode]           Specify the SPI mode to use (0-3).\n"
+           " spi-speed[=hertz]         Specify the SPI speed in hertz.\n"
+           " spi-cs-delay[=usec]       Specify the delay after C̅S̅ assertion, in µsec.\n"
+           " spi-reset-delay[=ms]      Specify the delay after R̅E̅S̅E̅T̅ assertion, in milliseconds.\n"
+           " spi-align-allowance[=n]   Specify the maximum number of 0xFF bytes to clip from start of\n"
+           "                           MISO frame. Max value is 16.\n"
+           " spi-small-packet=[n]      Specify the smallest packet we can receive in a single transaction.\n"
+           "                           (larger packets will require two transactions). Default value is 32.\n";
+#else
+    return "Radio URL:\n"
+           "spinel+hdlc+uart://${PATH_TO_UART_DEVICE}?${arguments} for real uart device\n"
+           "arguments:\n"
+           " uart-parity[=even|odd]     Uart parity config, optional.\n"
+           " uart-stop[=number-of-bits] Uart stop bit, default is 1.\n"
+           " baudrate[=baudrate]        Uart baud rate, default is 115200.\n"
+           "spinel+hdlc+fortpty://${PATH_TO_UART_DEVICE}?${arguments} for forking a pty subprocess.\n"
+           "arguments:\n"
+           " arg[=argument string]     Command line arguments, can be repeated.\n";
+#endif
+}
+
+namespace ot {
+namespace Posix {
+
+Arguments::Arguments(const char *aUrl)
+{
+    char *url = &mUrl[0];
+
+    VerifyOrDie(aUrl != NULL, OT_EXIT_INVALID_ARGUMENTS);
```

Comment:
How about adding a IsValid() method to validate the URL so that this utility doesn't rely on *Die()?